app/models/changeset/pull_request.rb

```diff
@@ -3,11 +3,14 @@ class Changeset::PullRequest
   RISKS_SECTION = /#+\s+Risks.*\n/i
 
   # Matches URLs to JIRA issues.
-  JIRA_ISSUE = %r[https://\w+\.atlassian\.net/browse/[\w-]+]
+  JIRA_ISSUE_URL = %r[https?:\/\/[\da-z\.\-]+\.[a-z\.]{2,6}\/browse\/[A-Z][A-Z\d]+-\d+]
 
   # Matches "VOICE-1234" or "[VOICE-1234]"
-  JIRA_CODE = %r[(\[)*([a-zA-Z]+)-((\d)+)(\])*]
+  JIRA_CODE_TITLE = %r[(\[)*([a-zA-Z]+)-((\d)+)(\])*]
 
+  # Matches "VOICE-1234" only
+  PUNCT = "\\s|\\p{Punct}|~|="
+  JIRA_CODE = %r[(?<=#{PUNCT}|^)([A-Z][A-Z\d]+-\d+)(?=#{PUNCT}|$)]
```

Comment:
Might be overkill on the amount of matching punctuation, but that's grand. :)  Should fix the comment as it matches all variants of VOICE-1234 using '"{}[]()=, so maybe point out a few examples.

The Project key matching part '[A-Z][A-Z\d]+' differs from the above JIRA_CODE_TITLE version '[a-zA-Z]+', so they should be made consistent.
Default Jira configuration has: '[A-Za-z][A-Za-z]+-\d+', though a jira admin can change it unfortunately. However, not worth the complexity of making it configurable IMO.